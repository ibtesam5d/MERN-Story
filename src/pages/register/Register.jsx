import React, { useState } from "react";
import "./Register.scss";
import { useNavigate } from "react-router-dom";
import newRequest from "../../utils/newRequest";
import upload from "../../utils/uploadImage";

const Register = () => {
  const [file, setFile] = useState(null);
  const [user, setUser] = useState({
    username: "",
    email: "",
    password: "",
    img: "",
    phone: "",
    isAuthor: false,
    desc: "",
  });

  const navigate = useNavigate();

  const handleChange = (e) => {
    setUser((prev) => {
      return { ...prev, [e.target.name]: e.target.value };
    });

    console.log(user);
  };

  const handleAuthor = (e) => {
    setUser((prev) => {
      return { ...prev, isAuthor: e.target.checked };
    });
  };
  const handleSubmit = async (e) => {
    e.preventDefault();

    const url = await upload(file);
    try {
      await newRequest.post("/auth/register", {
        ...user,
        img: url,
      });
      navigate("/");
    } catch (err) {
      console.log(err);
    }
  };

  return (
    <div className="register">
      <h1>Create a new account</h1>
      <form onSubmit={handleSubmit}>
        <div className="left">
          <label htmlFor="">Username</label>
          <input
            name="username"
            type="text"
            placeholder="johndoe"
            onChange={handleChange}
          />
          <label htmlFor="">Email</label>
          <input
            name="email"
            type="email"
            placeholder="email"
            onChange={handleChange}
          />
          <label htmlFor="">Password</label>
          <input name="password" type="password" onChange={handleChange} />
          <label htmlFor="">Profile Picture</label>
          <input type="file" onChange={(e) => setFile(e.target.files[0])} />
          <button type="submit">Register</button>
        </div>
        <div className="right">
          <h2>I want to become an author</h2>
          <div className="toggle">
            <label htmlFor="">Activate the author account</label>
            <label className="switch">
              <input type="checkbox" onChange={handleAuthor} />
              <span className="slider round"></span>
            </label>
          </div>
          <label htmlFor="">Phone Number</label>
          <input
            name="phone"
            type="text"
            placeholder="+1 234 567 89"
            onChange={handleChange}
          />
          <label htmlFor="">Description</label>
          <textarea
            placeholder="A short description of yourself"
            name="desc"
            id=""
            cols="30"
            rows="10"
            onChange={handleChange}
          ></textarea>
        </div>
      </form>
    </div>
  );
};

export default Register;
