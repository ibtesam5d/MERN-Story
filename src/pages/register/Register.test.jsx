import { render } from "@testing-library/react";
import { expect, it } from "vitest";
import { MemoryRouter } from "react-router-dom";
import Register from "./Register";

it("should match snapshot", () => {
  const route = "/register";
  const wrapperEl = render(
    <MemoryRouter initialEntries={[route]}>
      <Register />
    </MemoryRouter>
  );
  expect(wrapperEl).toMatchSnapshot();
});
