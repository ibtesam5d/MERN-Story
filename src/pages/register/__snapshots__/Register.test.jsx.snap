// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`should match snapshot 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="register"
      >
        <h1>
          Create a new account
        </h1>
        <form>
          <div
            class="left"
          >
            <label
              for=""
            >
              Username
            </label>
            <input
              name="username"
              placeholder="johndoe"
              type="text"
            />
            <label
              for=""
            >
              Email
            </label>
            <input
              name="email"
              placeholder="email"
              type="email"
            />
            <label
              for=""
            >
              Password
            </label>
            <input
              name="password"
              type="password"
            />
            <label
              for=""
            >
              Profile Picture
            </label>
            <input
              type="file"
            />
            <button
              type="submit"
            >
              Register
            </button>
          </div>
          <div
            class="right"
          >
            <h2>
              I want to become an author
            </h2>
            <div
              class="toggle"
            >
              <label
                for=""
              >
                Activate the author account
              </label>
              <label
                class="switch"
              >
                <input
                  type="checkbox"
                />
                <span
                  class="slider round"
                />
              </label>
            </div>
            <label
              for=""
            >
              Phone Number
            </label>
            <input
              name="phone"
              placeholder="+1 234 567 89"
              type="text"
            />
            <label
              for=""
            >
              Description
            </label>
            <textarea
              cols="30"
              id=""
              name="desc"
              placeholder="A short description of yourself"
              rows="10"
            />
          </div>
        </form>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="register"
    >
      <h1>
        Create a new account
      </h1>
      <form>
        <div
          class="left"
        >
          <label
            for=""
          >
            Username
          </label>
          <input
            name="username"
            placeholder="johndoe"
            type="text"
          />
          <label
            for=""
          >
            Email
          </label>
          <input
            name="email"
            placeholder="email"
            type="email"
          />
          <label
            for=""
          >
            Password
          </label>
          <input
            name="password"
            type="password"
          />
          <label
            for=""
          >
            Profile Picture
          </label>
          <input
            type="file"
          />
          <button
            type="submit"
          >
            Register
          </button>
        </div>
        <div
          class="right"
        >
          <h2>
            I want to become an author
          </h2>
          <div
            class="toggle"
          >
            <label
              for=""
            >
              Activate the author account
            </label>
            <label
              class="switch"
            >
              <input
                type="checkbox"
              />
              <span
                class="slider round"
              />
            </label>
          </div>
          <label
            for=""
          >
            Phone Number
          </label>
          <input
            name="phone"
            placeholder="+1 234 567 89"
            type="text"
          />
          <label
            for=""
          >
            Description
          </label>
          <textarea
            cols="30"
            id=""
            name="desc"
            placeholder="A short description of yourself"
            rows="10"
          />
        </div>
      </form>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
